src/highdicom/map/sop.py

```diff
@@ -0,0 +1,706 @@
+        referenced_series: Dict[str, List[Dataset]] = defaultdict(list)
+        for s_img in self._source_images:
+            ref = Dataset()
+            ref.ReferencedSOPClassUID = s_img.SOPClassUID
+            ref.ReferencedSOPInstanceUID = s_img.SOPInstanceUID
+            self.SourceImageSequence.append(ref)
+            referenced_series[s_img.SeriesInstanceUID].append(ref)
+
+        # Common Instance Reference
+        self.ReferencedSeriesSequence: List[Dataset] = []
+        for (
+            series_instance_uid,
+            referenced_images,
+        ) in referenced_series.items():
+            ref = Dataset()
+            ref.SeriesInstanceUID = series_instance_uid
+            ref.ReferencedInstanceSequence = referenced_images
+            self.ReferencedSeriesSequence.append(ref)
+
+        # Parametric Map Image
+        self.ImageType = ["DERIVED", "PRIMARY"]
+        self.LossyImageCompression = getattr(
+            src_img, 'LossyImageCompression', '00'
+        )
+        if self.LossyImageCompression == "01":
+            self.LossyImageCompressionRatio = (
+                src_img.LossyImageCompressionRatio
+            )
+            self.LossyImageCompressionMethod = (
+                src_img.LossyImageCompressionMethod
+            )
+        self.SamplesPerPixel = 1
+        self.PhotometricInterpretation = 'MONOCHROME2'
+        self.BurnedInAnnotation = 'NO'
+        if contains_recognizable_visual_features:
+            self.RecognizableVisualFeatures = 'YES'
+        else:
+            self.RecognizableVisualFeatures = 'NO'
+        self.ContentLabel = 'ISO_IR 192'  # UTF-8
+        self.ContentDescription = content_description
+        if content_creator_name is not None:
+            check_person_name(content_creator_name)
+        self.ContentCreatorName = content_creator_name
+        self.PresentationLUTShape = 'IDENTITY'
+
+        # TODO: check that physical dimensions of the image match those of the
+        # source image
+
+        self.DimensionIndexSequence = DimensionIndexSequence(coordinate_system)
+        dimension_organization = Dataset()
+        dimension_organization.DimensionOrganizationUID = (
+            self.DimensionIndexSequence[0].DimensionOrganizationUID
+        )
+        self.DimensionOrganizationSequence = [dimension_organization]
+
+        sffg_item = Dataset()
+
+        # If the same set of mappings applies to all frames, the information
+        # is stored in the Shared Functional Groups Sequence. Otherwise, it
+        # is stored for each frame separately in the Per Frame Functional
+        # Groups Sequence.
+        if pixel_array.ndim == 2:
+            pixel_array = pixel_array[np.newaxis, ..., np.newaxis]
+            sffg_item.RealWorldValueMappingSequence = real_world_value_mappings
+        elif pixel_array.ndim == 3:
+            pixel_array = pixel_array[..., np.newaxis]
+            sffg_item.RealWorldValueMappingSequence = real_world_value_mappings
```

Comment:
Need a check here on the size of the mappings list. What if the user has passed a Sequence[Sequence[RealWorldValueMapping]]?

Reply:
Addressed via 71f8940b0a303fb5d3695c9e3068ab05f644d053